cdap-ui/app/directives/complex-schema/complex-schema.js

```diff
@@ -127,10 +127,13 @@ function ComplexSchemaController (avsc, SCHEMA_TYPES, $scope, uuid, $timeout, Sc
       return;
     }
 
-    vm.formatOutput();
+    vm.formatOutput(true);
   }
 
-  vm.formatOutput = () => {
+  // In some cases, we edit the schema when the user opens a node, so the schema changes without the
+  // user doing anything. In those cases we should update the 'default' schema state to the state after
+  // we've done our initialzing i.e. updateDefault = true. Defaults to false.
+  vm.formatOutput = (updateDefault = false) => {
```

Comment:
Yes, it's used as a param of the `vm.parentFormatOutput` function that gets called inside this function.